### PR TITLE
Migrate to npm 7

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- enforce npm 15 and npm@7
- update all dependencies
- use webpack proxy for development
- fixed a bunch of invalid async tests that were creating actual API calls and crashing

### CODE REVIEW

**Review by karelhala — APPROVED** [review]

File: config/dev.webpack.config.js [hunk 1]
```diff
@@ -6,9 +6,10 @@ const config = require('@redhat-cloud-services/frontend-components-config');
 const { config: webpackConfig, plugins } = config({
   rootFolder: resolve(__dirname, '../'),
   debug: true,
-  https: false,
   sassPrefix: '.rbac, .my-user-access',
-  ...(process.env.BETA && { deployment: 'beta/apps' }),
+  deployment: process.env.BETA ? 'beta/apps' : 'apps',
+  useProxy: true,
+  appUrl: process.env.BETA ? ['/beta/settings/my-user-access', '/beta/settings/rbac'] : ['/settings/my-user-access', '/settings/rbac'],
```
Comment (karelhala) [inline config/dev.webpack.config.js:12]: Can you hide this behind `npm run start:proxy`? And document in README.md that we can run this either with webpack proxy or with insights proxy.

**Review by karelhala — CHANGES_REQUESTED** [review]

File: config/dev.webpack.config.js (hunk 1 above)
Comment (Hyperkid123) [reply config/dev.webpack.config.js:12]: Is this really necessary? There is no benefit to using insights proxy. We can achieve exactly the same result with webpack proxy and we can add routes config to customize it. Which is exactly the same as configuring the spandx

Comment (karelhala) [reply config/dev.webpack.config.js:12]: I still prefer using insights proxy (I have insights-proxy running on start with a lot of paths and just switch between app), at least have a task that runs this without webpack proxy.

Comment (Hyperkid123) [reply config/dev.webpack.config.js:12]: OK

**Review by karelhala — APPROVED** [review]

**Pull request merged**